### PR TITLE
BUGFIX: Skip node data processing for invalid properties

### LINKED ISSUE #4804: BUG: Prevent migration of invalid node properties
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

I tried to upgrade from Neos 8.3 to Neos 9.0. To do that, I upgraded neos-seo from 3.3.4 to 4.1.3 since this is the latest version and ran the migration using `cr:migrateLegacyData`. 

With the new changes in Neos, I get the following error while upgrading:

Bildschirmfoto 2023-12-14 um 10 21 49
This is because the twitter cards were removed in 4.1.2: https://github.com/neos/neos-seo/commit/41bade5db99335c24ddd41cde0064cb6b09f42ba 

### Expected Behavior

I expect to be able to use the latest version of neos-seo when migrating to Neos 9 or at least have some documentation that mentions that I should only user neos-seo 4.1.0 when migrating the database to Neos 9 .

### Steps To Reproduce

Upgrade from Neos 8.3 to 9.0 and have a twitter card information in the nodes

### Environment

```markdown
- Flow:9.0
- Neos:9.0
- PHP:8.2
```

### Anything else?

This is a copy of https://github.com/neos/neos-seo/issues/182

### PR DESCRIPTION
**Pull request by markusguenther**
It can happen that node properties are not available in a new version of the package. But when the properties are still part of the neos_contentrepository_domain_model_nodedata table, the migration tries to process the data for a node and throws an exception as the node property is not available in the node type schema anymore.

To prevent that exceptions in the legacy data migration, we skip the procession for such properties and output a warning for the user.

Fixes: #4804

**Review instructions**

As it is described in the linked issue, the twitter related properties in the Neos-Seo package have been removed in the latest version for Neos 9. When you upgrade your project from Neos 8.3 to Neos 9.0 you will increase the package version as well, but the data is still part of your database. To be able to test it, you can install a Neos 8.3 instance and change a Twitter Card property in the backend and publish that. When you now upgrade the Neos to version 9.0 and execute `./flow cr:migrateLegacyData` You should see something like that instead of an exception.
Screenshot 2023-12-15 at 18 04 02

**Checklist**

- [x] Code follows the PSR-2 coding style
- [ ] Tests have been created, run and adjusted as needed
- [x] The PR is created against the [lowest maintained branch](https://www.neos.io/features/release-roadmap.html)
- [ ] Reviewer - PR Title is brief but complete and starts with `FEATURE|TASK|BUGFIX`
- [ ] Reviewer - The first section explains the change briefly for change-logs
- [ ] Reviewer - Breaking Changes are marked with `!!!` and have upgrade-instructions

### CODE REVIEW

File: Neos.ContentRepository.LegacyNodeMigration/Classes/NodeDataToAssetsProcessor.php [hunk 1]
```diff
@@ -62,7 +62,12 @@ public function run(): ProcessorResult
                 continue;
             }
             foreach ($properties as $propertyName => $propertyValue) {
-                $propertyType = $nodeType->getPropertyType($propertyName);
+                try {
+                    $propertyType = $nodeType->getPropertyType($propertyName);
+                } catch (\InvalidArgumentException $exception) {
```
Comment (bwaidelich) [inline Neos.ContentRepository.LegacyNodeMigration/Classes/NodeDataToAssetsProcessor.php:67]: I don't see a corresponding `throw new \InvalidArgumentException` code in [NodeType::getPropertyType()](https://github.com/markusguenther/neos-development-collection/blob/731fc7fb8fe1d922a83b53218e948c51e206cca7/Neos.ContentRepository/Classes/Domain/Model/NodeType.php#L436) ?

File: Neos.ContentRepository.LegacyNodeMigration/Classes/NodeDataToEventsProcessor.php [hunk 2]
```diff
@@ -316,7 +316,12 @@ public function extractPropertyValuesAndReferences(array $nodeDataRow, NodeType
         }
 
         foreach ($decodedProperties as $propertyName => $propertyValue) {
-            $type = $nodeType->getPropertyType($propertyName);
+            try {
+                $type = $nodeType->getPropertyType($propertyName);
+            } catch (\InvalidArgumentException $exception) {
```
Comment (bwaidelich) [inline Neos.ContentRepository.LegacyNodeMigration/Classes/NodeDataToEventsProcessor.php:321]: same here

**Review by bwaidelich — COMMENTED** [review]
The fix makes sense to me in general, but I don't see how that should work.. A corresponding test would be nice

File: Neos.ContentRepository.LegacyNodeMigration/Classes/NodeDataToEventsProcessor.php (hunk 2 above)
Comment (markusguenther) [reply Neos.ContentRepository.LegacyNodeMigration/Classes/NodeDataToEventsProcessor.php:321]: Thanks for the feedback @bwaidelich, that comes from https://github.com/neos/neos-development-collection/blob/9.0/Neos.ContentRepository.Core/Classes/NodeType/NodeType.php in line 422.

Comment (bwaidelich) [reply Neos.ContentRepository.LegacyNodeMigration/Classes/NodeDataToEventsProcessor.php:321]: Ah of course, I looked at the wrong branch 👀 
Thanks for the clarification!

**Comment by bwaidelich** [comment]
Looking good, I'll try to add some tests

File: Neos.ContentRepository.LegacyNodeMigration/Classes/NodeDataToAssetsProcessor.php (hunk 1 above)
Comment (bwaidelich) [reply Neos.ContentRepository.LegacyNodeMigration/Classes/NodeDataToAssetsProcessor.php:67]: => https://github.com/neos/neos-development-collection/blob/9.0/Neos.ContentRepository.Core/Classes/NodeType/NodeType.php#L422-L425

**Review by bwaidelich — APPROVED** [review]
Thanks for taking care, I just added some tests

**Review by ahaeslich — APPROVED** [review]

**Pull request merged**